lib/ansible/plugins/action/reboot.py

```diff
@@ -220,16 +228,39 @@ def get_system_boot_time(self, distribution):
 
         display.debug("{action}: getting boot time with command: '{command}'".format(action=self._task.action, command=boot_time_command))
         command_result = self._low_level_execute_command(boot_time_command, sudoable=self.DEFAULT_SUDOABLE)
+        stdout = command_result['stdout']
+        stderr = command_result['stderr']
 
         if command_result['rc'] != 0:
-            stdout = command_result['stdout']
-            stderr = command_result['stderr']
             raise AnsibleError("{action}: failed to get host boot time info, rc: {rc}, stdout: {out}, stderr: {err}".format(
                                action=self._task.action,
                                rc=command_result['rc'],
                                out=to_native(stdout),
                                err=to_native(stderr)))
-        display.debug("{action}: last boot time: {boot}".format(action=self._task.action, boot=command_result['stdout'].strip()))
+
+        filter_re = self._get_value_from_facts('BOOT_TIME_FILTER_REGEX', distribution, 'DEFAULT_BOOT_TIME_FILTER_REGEX')
+        if self._task.args.get('boot_time_filter_regex'):
+            filter_re = self._task.args.get('boot_time_filter_regex')
+
+        display.debug("{action}: result of last boot time command: {stdout}".format(action=self._task.action, stdout=command_result['stdout'].strip()))
+
+        if filter_re:
+            matches = re.findall(filter_re, command_result['stdout'], re.MULTILINE)
+            if not matches:
+                raise AnsibleError(
+                    "{action}: failed to get host boot time info, filter regex "
+                    "did not match result of boot time check command. rc: {rc}, "
+                    "stdout: {out}, stderr: {err}".format(
+                        action=self._task.action,
+                        rc=command_result['rc'],
+                        out=to_native(stdout),
+                        err=to_native(stderr)))
+
+            # If there are multiple matches, munge them together into a string
+            # We could return a list, but we don't actually use the data IN the
+            # result for anything other than seeing if it changed, so
+            # constructing a string works fine.
+            return '\n'.join(matches)
```

Comment:
I think there's still a race where the line we're looking for won't necessarily be the last line of the output -- where it could output that line, but still capture the shutdown notice before the program exits.

Reply:
True.

Reply:
last line wont work, but you can at least rely on the output of the command sending it's own line break